tests/integrationv2/test_signature_algorithms.py

```diff
@@ -176,7 +183,72 @@ def test_s2n_client_signature_algorithms(managed_process, cipher, provider, othe
         results.assert_success()
         assert to_bytes("Actual protocol version: {}".format(
             expected_version)) in results.stdout
-        assert signature_marker(
-            Provider.ServerMode, signature) in results.stdout or not server_sigalg_used
-        assert (signature_marker(Provider.ClientMode, signature)
-                in results.stdout) == client_auth
+        # TLS1.1 always negotiates to RSA+MD5_SHA1 signature algorithm.
+        if protocol < Protocols.TLS12:
+            assert signature_marker(
+                Provider.ServerMode, Signatures.MD5_SHA1
+            ) in results.stdout
+            assert (signature_marker(Provider.ClientMode, Signatures.MD5_SHA1)
+                    in results.stdout) == client_auth
+        else:
+            assert signature_marker(
+                Provider.ServerMode, signature) in results.stdout or not server_sigalg_used
+            assert (signature_marker(Provider.ClientMode, signature)
+                    in results.stdout) == client_auth
+
+
+@pytest.mark.uncollect_if(func=skip_ciphers)
+@pytest.mark.parametrize("cipher", [Ciphers.SECURITY_POLICY_20210816], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL, GnuTLS])
+@pytest.mark.parametrize("protocol", [Protocols.TLS12], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
+# RSA_SHA224 signature algorithm is not included in security_policy_20210816[].
+@pytest.mark.parametrize("signature", [Signatures.RSA_SHA224], ids=get_parameter_name)
```

Comment:
I think this signature algorithm isn't compatible with gnutls. gnutls checks for compatibility here:
https://github.com/aws/s2n-tls/blob/f4b8afec58a2c199e0349b3f238c50229e007605/tests/integrationv2/providers.py#L681-L688

I know I suggested adding gnutls, but thinking about it now, I'm not sure adding it really helps this test very much, and it might complicate the test for no reason. This test is ensuring that the s2n-tls server is able to fallback to a default signature algorithm if there's a mismatch. As long as that functionality is triggered via openssl, I'm not sure triggering it with another library helps test anything really. So, I would either set a signature algorithm that gnutls supports (and is a mismatch with the security policy) or maybe just remove gnutls as a provider.